kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/KieServerInstanceManager.java

```diff
@@ -150,83 +148,113 @@ public Void doOperation(KieServicesClient client,
-                                                            configItem.setName(KieServerConstants.PCFG_RUNTIME_STRATEGY);
-                                                            configItem.setValue(processConfig.getRuntimeStrategy());
-
-                                                            containerResource.addConfigItem(configItem);
-                                                        }
-                                                    }
+        final RemoteKieServerOperation<Void> startContainerOperation = makeStartContainerOperation(containerSpec);
 
-                                                    ServiceResponse<KieContainerResource> response = client.createContainer(containerSpec.getId(),
-                                                                                                                            containerResource);
-                                                    if (!response.getType().equals(ServiceResponse.ResponseType.SUCCESS)) {
-                                                        logger.debug("Container {} failed to start on server instance {} due to {}",
-                                                                     containerSpec.getId(),
-                                                                     container.getUrl(),
-                                                                     response.getMsg());
-                                                    }
-                                                    collectContainerInfo(containerSpec,
-                                                                         client,
-                                                                         container);
-                                                    return null;
-                                                }
-                                            });
+        return callRemoteKieServerOperation(serverTemplate, containerSpec, startContainerOperation);
+    }
+
+    RemoteKieServerOperation<Void> makeStartContainerOperation(final ContainerSpec containerSpec) {
+        return new RemoteKieServerOperation<Void>() {
+            @Override
+            public Void doOperation(final KieServicesClient client,
+                                    final Container container) {
+
+                final KieContainerResource resource = makeContainerResource(container, containerSpec);
+                final ServiceResponse<KieContainerResource> response = client.createContainer(containerSpec.getId(), resource);
+
+                if (response.getType() != ServiceResponse.ResponseType.SUCCESS) {
+                    log("Container {} failed to start on server instance {} due to {}", container, response, containerSpec);
+                }
+
+                collectContainerInfo(containerSpec, client, container);
+
+                return null;
+            }
+        };
+    }
+
+    void log(final String message,
```

Comment:
this method looks kind of weird as it include just one line inside, most likely not really needed, isn't it?

Reply:
@mswiderski I extract to a method, to verify (via unit tests) some behaviors, like this one:
https://github.com/karreiro/droolsjbpm-integration/blob/6af1b92b91b930caf3b6dae130a29b426bb8c023/kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/KieServerInstanceManager.java#L168:L170
Since `logger` is declared like this:
```java
private static final Logger logger
```
I think that this is a good (and maybe weird) way to verify this logic :-)

Reply:
fair enough, if there is a good reason for it then I am fine with it :)